tests/integration/cocoon/test_cocoon_holdings.py

```diff
@@ -725,66 +746,146 @@ def test_load_from_data_frame_holdings_success(
+            ],
+            [
+                "One portfolio, one sub-holding-key, one date, two instruments",
+                "data/holdings-example-one-portfolio-one-shk-one-date-two-instruments.csv",
+                ["Security Description"],
+                False,
+                holding_instrument_identifiers(fake1=True, fake2=True),
+            ],
+            [
+                "One portfolio, one sub-holding-key, two dates, two instruments",
+                "data/holdings-example-one-portfolio-one-shk-two-dates-two-instruments.csv",
+                ["Security Description"],
+                False,
+                holding_instrument_identifiers(fake1=True, fake2=True),
+            ],
+            [
+                "One portfolio, two sub-holding-keys, one date, same instrument",
+                "data/holdings-example-one-portfolio-two-shks-one-date-one-instrument.csv",
+                ["Security Description"],
+                False,
+                holding_instrument_identifiers(fake1=True, fake2=False),
+            ],
+            [
+                "One portfolio, two sub-holding-keys, one date, two instruments",
+                "data/holdings-example-one-portfolio-two-shks-one-date-two-instruments.csv",
+                ["Security Description"],
+                False,
+                holding_instrument_identifiers(fake1=True, fake2=True),
+            ],
+            [
+                "One portfolio, two sub-holding-keys, two dates, two instruments",
+                "data/holdings-example-one-portfolio-two-shks-two-dates-two-instruments.csv",
+                ["Security Description"],
+                False,
+                holding_instrument_identifiers(fake1=True, fake2=True),
+            ],
+            [
+                "One portfolio, two sub-holding-keys, two dates, same instrument",
+                "data/holdings-example-one-portfolio-two-shks-two-dates-one-instrument.csv",
+                ["Security Description"],
+                False,
+                holding_instrument_identifiers(fake1=True, fake2=False),
+            ],
+            [
+                "One portfolio, two sub-holding-keys, two dates, same instrument - AdjustHolding",
+                "data/holdings-example-one-portfolio-two-shks-two-dates-one-instrument.csv",
+                ["Security Description"],
                 True,
-                [
-                    {
-                        "Instrument/default/Sedol": "9088840066",
-                        "Instrument/default/Isin": "US02HF9234H67",
-                    },
-                    {
-                        "Instrument/default/Sedol": "BFAKE712",
-                        "Instrument/default/Isin": "USFAHHDAR1014",
-                    },
-                ],
+                holding_instrument_identifiers(fake1=True, fake2=False),
+            ],
+            [
+                "One portfolio, one sub-holding-key, two dates, two instruments - AdjustHolding",
```

Comment:
are these all SetHoldings? I think we should add explicitly whether each one is Set or Adjust

Reply:
Good point